File: gems/decomplex/examples/go/structural-topology.go
package main
type Worker struct {}
func (self *Worker) run(items Items) { self.prepare(); if self.ready() { self.validate() }; for _, item := range items { self.helper(item) } }
func (self *Worker) prepare() {}
func (self *Worker) ready() bool { return true }
func (self *Worker) validate() {}
func (self *Worker) helper(item Item) { item.use() }

File: gems/decomplex/examples/kotlin/structural-topology.kt
class Worker {
  fun run(items: Items) {
    this.prepare()
    if (this.ready()) {
      this.validate()
    }
    for (item in items) {
      this.helper(item)
    }
  }

  private fun prepare() {}
  private fun ready(): Boolean { return true }
  fun validate() {}
  private fun helper(item: Item) { item.use() }
}
